packages/shared/lib/services/connections/utils.ts

```diff
@@ -0,0 +1,25 @@
+import ms from 'ms';
+import type { AllAuthCredentials, DBConnection, DBConnectionAsJSONRow } from '@nangohq/types';
+
+export const DEFAULT_EXPIRES_AT_MS = ms('1day');
+export const DEFAULT_OAUTHCC_EXPIRES_AT_MS = ms('55minutes'); // This ensures we have an expiresAt value
+export const DEFAULT_BILL_EXPIRES_AT_MS = ms('35minutes'); //This ensures we have an expireAt value for Bill
+export const DEFAULT_UNAUTH_EXPIRES_AT_MS = ms('99years');
+export const MAX_FAILED_REFRESH = 30;
+export const REFRESH_MARGIN_S = ms('15minutes') / 1000;
+
+export function getExpiresAtFromCredentials(credentials: AllAuthCredentials): Date | null {
+    if ('expires_at' in credentials && credentials['expires_at']) {
+        return credentials['expires_at'];
+    }
+
+    if (credentials.type === 'CUSTOM' || credentials.type === 'OAUTH1' || credentials.type === 'APP_STORE' || !credentials.type) {
+        return new Date(Date.now() + DEFAULT_UNAUTH_EXPIRES_AT_MS);
```

Comment:
name of the constant is a bit confusing. What about mentioning infinite expiration or something similar?